jbi/jira/service.py

```diff
@@ -368,229 +367,182 @@ def update_issue_labels(
-                id="jira.components.parsing",
-            )
-        ]
-
-    unknown = specified_components - all_components_names
-    if unknown:
-        return [
-            checks.Warning(
-                f"Jira project {project_key} does not have components {unknown}",
-                id="jira.components.missing",
-            )
-        ]
-
-    return []
-
-
-def check_jira_all_project_issue_types_exist(actions, _get_service):
-    # Do not bother executing the rest of checks if connection fails.
-    service = _get_service()
-
-    if messages := check_jira_connection(_get_service):
-        return messages
-
-    try:
-        paginated_project_response = service.client.paginated_projects(
-            expand="issueTypes", keys=actions.configured_jira_projects_keys
-        )
-    except requests.RequestException:
-        return [
-            checks.Error(
-                "Couldn't fetch projects",
-                id="jira.projects.error",
+        missing_issue_types_by_project = {}
+        for action in actions:
+            action_issue_types = set(action.parameters.issue_type_map.values())
+            project_issue_types = issue_types_by_project.get(
+                action.jira_project_key, set()
             )
-        ]
-
-    projects = paginated_project_response["values"]
-    issue_types_by_project = {
-        project["key"]: {issue_type["name"] for issue_type in project["issueTypes"]}
-        for project in projects
-    }
-    missing_issue_types_by_project = {}
-    for action in actions:
-        action_issue_types = set(action.parameters.issue_type_map.values())
-        project_issue_types = issue_types_by_project.get(action.jira_project_key, set())
-        if missing_issue_types := action_issue_types - project_issue_types:
-            missing_issue_types_by_project[action.jira_project_key] = (
-                missing_issue_types
-            )
-    if missing_issue_types_by_project:
-        return [
-            checks.Warning(
-                f"Jira projects {set(missing_issue_types_by_project.keys())} with missing issue types",
-                obj=missing_issue_types_by_project,
-                id="jira.types.missing",
-            )
-        ]
-    return []
+            if missing_issue_types := action_issue_types - project_issue_types:
+                missing_issue_types_by_project[action.jira_project_key] = (
+                    missing_issue_types
+                )
+        if missing_issue_types_by_project:
+            return [
+                checks.Warning(
+                    f"Jira projects {set(missing_issue_types_by_project.keys())} with missing issue types",
+                    obj=missing_issue_types_by_project,
+                    id="jira.types.missing",
+                )
+            ]
+        return []
 
+    def check_jira_pandoc_install(self):
+        if markdown_to_jira("- Test") != "* Test":
```

Comment:
I would have assumed we'd get an exception rather than an invalid string being returned.

Reply:
Hmm, you made me doubt... if an exception is raised then the endpoint returns 500 as expected